en/sql/tuning.rst

```diff
@@ -3829,6 +3829,89 @@ The join with the *right_tbl* table was eliminated.
+
+However, with **Predicate Push**, if the query is transformed as follows, the optimization ensures that only fewer amounts of data are queried.
+
+.. code-block:: sql
+
+        SELECT a.name, r.score 
+        FROM (SELECT name, nation_code, code, count(*) cnt FROM athlete WHERE nation_code = 'KOR' GROUP BY name, nation_code ) a, record r
+        WHERE a.code = r.athlete_code;
+
+In CUBRID, **Predicate Push** can't be performed if the query meets the following conditions:
+
+#. Using **NO_PUSH_PRED** hint on main query.
+
+#. Contains **CONNECT BY**.
+    
+#. Using aggregation or analytical functions.
+
+#. Using **ROWNUM, LIMIT**, or **GROUPBY_NUM(), INST_NUM(), ORDERBY_NUM()**.
+    
+#. Written using **Correlated Subquery**.
+
+#. When subqueries are used in predicates.
+
+#. The view includes methods.
+
+#. When the predicate to be pushed or the target for **Predicate Push** within the view uses **RANDOM (), DRANDOM (), SYS_GUID ()**\.
+
+#. When performing an **OUTER JOIN** and either the predicate to be pushed or the target for **Predicate Push** within the view uses:
+        * Predicates written in the **ON** clause.
+        * **NULL** transformation functions.
+
+The following is an example that uses **NO_PUSH_PRED** hint on main query.
+
+.. code-block:: sql
+
+        SELECT /*+ NO_PUSH_PRED*/ a.name, r.score 
+        FROM (SELECT name, nation_code, code, count(*) cnt FROM athlete GROUP BY name, nation_code) a, record r
+        WHERE a.code = r.athlete_code
+        AND a.nation_code = 'KOR';
+
+If the **NO_PUSH_PRED** hint is used on main query, **Predicate Push** is not applied.
+
+The following is an example that performs an **OUTER JOIN** with the predicate to be pushed in the **ON** clause condition.
+
+.. code-block:: sql
+
+        SELECT a.name, r.score 
+        FROM (SELECT * FROM athlete WHERE gender = 'M') a
+            LEFT OUTER JOIN record r ON a.code = r.athlete_code AND a.nation_code = 'KOR';
+
+In this case, *a.nation_code = 'KOR'* exists in the **ON** clause during the **LEFT OUTER JOIN**. Such a predicate within the **ON** clause is not a target for **Predicate Push**.
+
+The following is an example that performs an **OUTER JOIN** where either the predicate to be pushed or the target for **Predicate Push** within the view uses the **NULL** transformation function.
+
+.. code-block:: sql
+
+        SELECT a.name, r.score 
+        FROM athlete a
+                LEFT OUTER JOIN (SELECT * FROM record WHERE medal = 'G') r ON a.code = r.athlete_code
+        WHERE NVL(r.score, '0') = '0';
+
+When performing an **OUTER JOIN** and either the predicate to be pushed or the target for **Predicate Push** within the view uses a **NULL** transformation function, it's not a target for **Predicate Push**. This includes **COALESCE (), NVL (), NVL2 (), DECODE (), IF (), IFNULL (), CONCAT_WS ()**. Furthermore, **IS NULL ,CASE** statements are also not targets for **Predicate Push**.
```

Comment:
I think it would be clearer to list the null-related functions in the description above rather than in the example.

> This includes **COALESCE (), NVL (), NVL2 (), DECODE (), IF (), IFNULL (), CONCAT_WS ()**.

Reply:
I commented on the change in location.

Reply:
I commented on the change in location.

Reply:
And I think that `resolve the conversation` should be done by the comment writer after checking the results.
Alternatively, the review requester can `resolve the conversation` by receiving a check from the comment writer.

Reply:
Please understand that my expression may not have been clear.

I wrote the below in the top comment.
> I think it would be clearer to list the null-related functions in the description above rather than in the example.

I think there should be a list of the null-related functions along with the `**NULL** transformation functions.` sentence.